File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/access/annotationprovider/DefaultMessageProvider.java
package ru.tggc.telegrambotcore.access.annotationprovider;

import org.springframework.stereotype.Component;
import ru.tggc.telegrambotcore.annotation.handle.DefaultMessageHandle;
import ru.tggc.telegrambotcore.annotation.handle.HandleMeta;

import java.lang.reflect.Method;

@Component
public class DefaultMessageProvider implements AnnotationProvider {

    @Override
    public boolean supports(Method m) {
        return m.isAnnotationPresent(DefaultMessageHandle.class);
    }

    @Override
    public HandleMeta extractMeta(Method m) {
        return HandleMeta.fromDefault();
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/access/annotationprovider/DefaultMessageProvider.kt
package ru.tggc.telegrambotcore.access.annotationprovider

import org.springframework.stereotype.Component
import ru.tggc.telegrambotcore.annotation.handle.DefaultMessageHandle
import ru.tggc.telegrambotcore.annotation.handle.HandleMeta
import java.lang.reflect.Method

@Component
class DefaultMessageProvider : AnnotationProvider {
    override fun supports(m: Method): Boolean = m.isAnnotationPresent(DefaultMessageHandle::class.java)


    override fun extractMeta(m: Method): HandleMeta? = HandleMeta.fromDefault()

}
